linkerd/app/integration/src/tests/telemetry/tls_detect.rs

```diff
@@ -0,0 +1,124 @@
+use super::*;
+use std::time::SystemTime;
+
+const TIMEOUT: Duration = Duration::from_millis(640); // 640ms ought to be enough for anybody.
+
+/// A helper that builds a proxy with the above detect timeout and a TCP server that always drops
+/// the accepted socket.
+async fn run_proxy() -> proxy::Listening {
+    let identity::Identity {
+        mut env,
+        mut certify_rsp,
+        ..
+    } = identity::Identity::new(
+        "foo-ns1",
+        "foo.ns1.serviceaccount.identity.linkerd.cluster.local".to_string(),
+    );
+
+    // The identity service is needed for the proxy to start.
```

Comment:
hmm, in this case, it's needed specifically because we want to test behavior with TLS enabled --- so i think this should maybe be something more like
```rust
// The identity service is required for the proxy to start because identity is
// enabled in these tests
```

but, not a big deal.

Reply:
Yeah, please feel free to correct these comments, I was just taking a stab at explaining it